sapi/litespeed/lsapilib.c

```diff
@@ -1899,6 +1909,40 @@ ssize_t gsendfile( int fdOut, int fdIn, off_t* off, size_t size )
 }
 #endif
 
+#if defined(__OpenBSD__)
+ssize_t gsendfile( int fdOut, int fdIn, off_t* off, size_t size )
+{
+    ssize_t ret;
+    off_t written = 0;
+    const size_t bufsiz = 16384;
+    unsigned char in[bufsiz] = {0};
+
+    if (lseek(fdIn, *off, SEEK_SET) == -1) {
+        return -1;
+    }
+
+    while (size > 0) {
+	    size_t tor = size > sizeof(in) ? sizeof(in) : size;
+	    ssize_t c = read(fdIn, in, tor);
+	    if (c <= 0) {
+		    goto end;
+	    }
+
+	    ssize_t w = write(fdOut, in, c);
+	    if (w != c) {
+		    goto end;
+	    }
+
+	    written += w;
+	    size -= c;
+    }
+
+end:
+    *off += written;
+    return 0;
+}
+#endif
```

Comment:
Is this sendfile implementation taken from somewhere?

Reply:
Not taken, in the first this is just read a file descriptor and writing in another one, I chose a small buffer for starter then increased to gain a little bit but no match for kernel space implementation.